c10/core/WrapDimMinimal.cpp

```diff
@@ -3,7 +3,7 @@
 namespace c10::detail {
 
 template <typename T>
-T maybe_wrap_dim_slow(T dim, T dim_post_expr, bool wrap_scalar) {
+T maybe_wrap_dim_slow(T dim, const T& dim_post_expr, bool wrap_scalar) {
```

Comment:
Why are these changes needed? Were they overshadowing other overloads? It's a bit of a surprise why only the second value is const ref so it might be good to add a comment.

Reply:
Looks like it might add about 4-5 minutes max to clang analyzer or 25% overhead which is acceptable.

Reply:
These are some fixed detected by the new checks.

Reply:
> Why are these changes needed? Were they overshadowing other overloads? It's a bit of a surprise why only the second value is const ref so it might be good to add a comment.

These are some fixed detected by the new checks.